### DISCUSSION TITLE
skynet 启动两个不同的独立的进程服务，集群配置harbor=0，相互关联的通信服务地址已经reload但是无法连接上，请问云风大佬可能引起无法连接的原因有哪些

### DISCUSSION THREAD

**Discussion by Smdocc** [cloudwu/skynet#1515] [General]

**Comment by Smdocc** [comment]
### 测试代码：
```lua
function ServerStart.StartA()
    local serverA = "serverA"
    local serverB = "serverB"
    local clusterMap = {}
    clusterMap[serverA] = "192.168.22.68:2100"
    clusterMap[serverB] = "192.168.22.68:2200"
    cluster.reload(clusterMap)
    cluster.open(serverA)
    local srvmanagerserviceA = skynet.newservice("srvmanagerserviceA")
    skynet.name(".srvmanagerserviceA", srvmanagerserviceA)
    cluster.register("srvmanagerserviceA", srvmanagerserviceA)
    skynet.sleep(600)
    cluster.send(serverB, "@srvmanagerserviceB", "testcode", {
        a = 1
    })
end

function ServerStart.StartB()
    local serverA = "serverA"
    local serverB = "serverB"
    local clusterMap = {}
    clusterMap[serverA] = "192.168.22.68:2100"
    clusterMap[serverB] = "192.168.22.68:2200"
    cluster.reload(clusterMap)
    cluster.open(serverB)
    local srvmanagerserviceB = skynet.newservice("srvmanagerserviceB")
    skynet.name(".srvmanagerserviceB", srvmanagerserviceB)
    cluster.register("srvmanagerserviceB", srvmanagerserviceB)
    skynet.sleep(600)
    cluster.send(serverA, "@srvmanagerserviceA", "testcode", {
        a = 1
    })
end
```

**ServerStart.StartA() 和ServerStart.StartB() 分别在对应的main中执行 启动配置harbor=0**

**Reply by dingdalong** [reply]
` ServerStart.StartB()`中，`serverB`地址的端口需要和`ServerStart.StartA()`的相同

**Reply by Smdocc** [reply]
**相同的已经测试过了依旧收不到对方发送的消息**

**这是B服务的启动信息
**
[:00000002] LAUNCH snlua bootstrap
[:00000003] LAUNCH snlua launcher
[:00000004] LAUNCH snlua cdummy
[:00000005] LAUNCH harbor 0 4
[:00000006] LAUNCH snlua datacenterd
[:00000007] LAUNCH snlua service_mgr
[:00000008] LAUNCH snlua main
[:00000009] LAUNCH snlua clusterd
[:0000000a] LAUNCH snlua gate
[:0000000a] Listen on 192.168.22.68:2200
[:0000000b] LAUNCH snlua srvmanagerserviceB
[:00000009] Register [srvmanagerserviceB] :0000000b
**

**这是A服务的启动信息**
**
[:00000002] LAUNCH snlua bootstrap
[:00000003] LAUNCH snlua launcher
[:00000004] LAUNCH snlua cdummy
[:00000005] LAUNCH harbor 0 4
[:00000006] LAUNCH snlua datacenterd
[:00000007] LAUNCH snlua service_mgr
[:00000008] LAUNCH snlua main
[:00000009] LAUNCH snlua clusterd
[:0000000a] LAUNCH snlua gate
[:0000000a] Listen on 192.168.22.68:2100
[:0000000b] LAUNCH snlua srvmanagerserviceA
[:00000009] Register [srvmanagerserviceA] :0000000b
[:00000008] KILL self
[:00000002] KILL self
**

**Reply by dingdalong** [reply]
`A`服务中，调用了两次`cluster.register("srvmanagerserviceA", srvmanagerserviceA)`，你那边不会报错？
我这边按你的逻辑跑了一下测试，是ok的。你可以用`telnet`连一下`192.168.22.68:2100`看看能否连通。
最后还可以看看`examples`中的`cluster`的例子是否能跑通。

**Reply by Smdocc** [reply]
抱歉，A服那个是粘贴代码时的问题， 服务能正常启动， telnet 也能收到消息响应但是就是发送目标服务的时候无法收到响应

**Reply by Smdocc** [reply]
谢谢， 我再看看examples 中的cluster例子

**Reply by dingdalong** [reply]
加油~

**Reply by Smdocc** [reply]
**刚刚发现 通过本地代理** 
   ``` 
   local proxy = cluster.proxy "serverA@srvmanagerserviceA"
    skynet.send(proxy, "lua", "testcode") 
``` 
**发送的话可以收到相应的响应但是通过下调用下面的代码却收不到响应消息**
```
    cluster.send("serverA", "@srvmanagerserviceA", "testcode")
```
**有点让人不解**

**Reply by Gowa2017** [reply]
你看看 @ 是什么意思

**Comment by cloudwu** [comment]
学习一下如何用 markdown 贴代码先：https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

**Reply by Smdocc** [reply]
测试代码重新上传了， 新手见谅
